exp/ipam/api/v1alpha1/ipaddress_types.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IPAddressSpec is the desired state of an IPAddress.
+type IPAddressSpec struct {
+	// ClaimRef is a reference to the claim this IPAddress was created for.
+	ClaimRef corev1.LocalObjectReference `json:"claimRef,omitempty"`
```

Comment:
What is the value of using omitempty in a struct which does not have empty value?

Reply:
Same question for the other fields.

Reply:
It's also specified in other places, e.g. the Cluster spec. But there the `+optional` flag is set as well.
According to the Go docs, `omitempty` is pointless on structs as you mentioned. It does work for pointers though.
So we can probably remove it in a lot more places. On the other hand it doesn't do any harm, so it might not be worth the effort of removing it everywhere where it's unused.

Reply:
aside from moving forward with the PR, does it actually makes any difference? how does a client could not set the struct given is not a pointer?